### DISCUSSION TITLE
[开发]PyQt5是怎么在PyQt6.5.2上运行的？

### DISCUSSION THREAD

**Discussion by Andy-AO** [bookfere/Ebook-Translator-Calibre-Plugin#273] [Q&A] [answered]
在命令行中运行py代码，发现Calibre内部的PyQt运行环境是PyQt6.5.2，直接在插件中插入代码执行也是相同结果
```log
 calibre-debug -c "import PyQt5.QtCore;print('PyQt5 version:');print(PyQt5.QtCore.PYQT_VERSION_STR);print('python sys.version:');import sys;print(sys.version)"
PyQt5 version:
WARNING: CPU random generator seem to be failing, disabling hardware random number generation
WARNING: RDRND generated: 0xffffffff 0xffffffff 0xffffffff 0xffffffff
6.5.2
python sys.version:
3.11.5 (main, Oct  9 2023, 08:50:02) [MSC v.1937 64 bit (AMD64)]
```
这个有点奇怪，明明调用的是PyQt5，版本号出来是6.5.2，这是怎么回事？
插件用的到底是PyQt5还是PyQt6？

**Comment by bookfere** [comment, marked as answer]
[Calibre 从版本 6 开始使用 PyQt6](https://www.mobileread.com/forums/showthread.php?t=344064)，新版本 Calibre 中的 PyQt5 是为兼容第三方插件[添加的别名](https://github.com/kovidgoyal/calibre/blob/7e5f328fc37bf8418e8b4515b87f57562f5af067/src/calibre/constants.py#L213-L217)。
